### PR TITLE
GH-38925: [CI] Fix spelling

### LINKED ISSUE #38925: [CI] Spelling errors identified by check-spelling
### Describe the bug, including details regarding any error messages, version, and platform.

The [check-spelling action](https://github.com/marketplace/actions/check-spelling) enabled me to identify some misspellings...

The misspellings have been reported at https://github.com/jsoref/apache-arrow/actions/runs/7008141484/attempts/1#summary-19063787889

### Component(s)

Continuous Integration

### PR DESCRIPTION
**Pull request by jsoref**
### What changes are included in this PR?

Spelling fixes to ci/

### Are there any user-facing changes?

* Closes: #38925

### CODE REVIEW

File: ci/conan/all/conanfile.py [hunk 1]
```diff
@@ -328,7 +328,7 @@ def _with_boost(self, required=False):
             return bool(self.options.with_boost)
 
     def _with_thrift(self, required=False):
-        # No self.options.with_thift exists
+        # No self.options.with_thrift exists
```
Comment (kou) [inline ci/conan/all/conanfile.py:331]: Could you report this to the upstream?
https://github.com/conan-io/conan-center-index/blob/master/recipes/arrow/all/conanfile.py

File: ci/conan/all/test_package/test_package.cpp [hunk 2]
```diff
@@ -78,7 +78,7 @@ arrow::Status VectorToColumnarTable(const std::vector<struct data_row>& rows,
     // Indicate the start of a new list row. This will memorise the current
     // offset in the values builder.
     ARROW_RETURN_NOT_OK(components_builder.Append());
-    // Store the actual values. The final nullptr argument tells the underyling
+    // Store the actual values. The final nullptr argument tells the underlying
```
Comment (kou) [inline ci/conan/all/test_package/test_package.cpp:81]: Could you report this to the upstream?
https://github.com/conan-io/conan-center-index/blob/master/recipes/arrow/all/test_package/test_package.cpp

**Review by kou — APPROVED** [review]
+1

File: ci/conan/all/conanfile.py (hunk 1 above)
Comment (jsoref) [reply ci/conan/all/conanfile.py:331]: I've added the repository to my to-do list, but I'm already a number of levels upstream. I can try to do it once I clear out the rest of these splits... 


(I did Google FlatBuffers)

**Pull request merged**